libs/components/modals/src/lib/modules/confirm/confirm.component.ts

```diff
@@ -70,99 +70,66 @@ export class SkyConfirmComponent {
   #getPresetButtons(): Observable<SkyConfirmButton[]> {
     const emitter = new ReplaySubject<SkyConfirmButton[]>(1);
 
-    switch (this.#config.type) {
```

Comment:
I saw some warnings in this code, so while addressing them I took the opportunity to condense some of the repetitive code.